src/main/java/ch/njol/skript/expressions/ExprEntityAttribute.java

```diff
@@ -0,0 +1,122 @@
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.attribute.Attribute;
+
+import java.util.stream.Stream;
+
+import org.bukkit.attribute.Attributable;
+import org.bukkit.attribute.AttributeInstance;
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Entity Attribute")
+@Description({"The attribute value of an entity.",
+			 "Note that the movement speed attribute cannot be reliably used for players. For that purpose, use the speed expression instead."})
+@Examples({"on damage of player:",
+		"	send \"You are wounded!\"",
+		"	set victim's attack speed attribute to 2"})
+@Since("INSERT VERSION")
+public class ExprEntityAttribute extends PropertyExpression<Entity, Number> {
+	
+	static {
+		Skript.registerExpression(ExprEntityAttribute.class, Number.class, ExpressionType.COMBINED,
+				"%attributetype% [value] of %entities%",
+				"%entities%'[s] %attributetype% [value]");
+	}
+	
+	@Nullable
+	private Expression<Attribute> attributes;
+	
+	@SuppressWarnings({"null", "unchecked"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		attributes = (Expression<Attribute>) exprs[matchedPattern];
+		setExpr((Expression<? extends Entity>) exprs[matchedPattern ^ 1]);
+		return true;
+	}
+
+	@SuppressWarnings("null")
+	@Override
+	protected Number[] get(Event e, Entity[] entities) {
+		Attribute a = attributes.getSingle(e);
+		return Stream.of(entities)
+		    .map(ent -> getAttribute(ent, a).getBaseValue())
+		    .toArray(Number[]::new);
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET)
+			return CollectionUtils.array(Number.class);
+		return null;
+	}
+
+	@SuppressWarnings("null")
+	@Override
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+		Attribute a = attributes.getSingle(e);
+		double d = ((Number) delta[0]).doubleValue();
+		if (mode == ChangeMode.SET) {
+			for (Entity entity : getExpr().getArray(e)) {
+				AttributeInstance ai = getAttribute(entity, a);
+				if (ai != null) {
+					ai.setBaseValue(d);
+				}
+			}
+		}
+	}
+
+	@Override
+	public Class<? extends Number> getReturnType() {
+		return Number.class;
+	}
+
+	@SuppressWarnings("null") // For some reason Java thinks that entities.toString(...) is null even though there's a null check ternary 
```

Comment:
you can go without the suppresswarnings by setting a local var to the expr in the method, then using that. skript does that in some places and suppresses in others ¯\_(ツ)_/¯

Reply:
Also btw it's not an issue with Java but rather the, well, a little "non-standard" compiler we use, which is ECJ, the Eclipse Compiler for Java. And by default it doesn't behave like that, we have it configured to care about nulls so much for some reason... and in this particular case, it complains because *theoretically* in the meantime between executing `attributes == null` and `attributes.toString(e, debug))`, another thread could set `attributes` to null. 

But that's just theory, and in most cases where the compiler complains about nulls we know it will never happen so we can suppress the warning. Or like @Pikachu920 says, we can introduce a local variable that other threads can't just access, so the issue will disappear.

I'm not a fan of using ECJ at all but that's the reasoning :)

Reply:
Thanks for the explanation, I have removed my comment in the next commit